File: realtube-python/app/models/video.py
from datetime import datetime

from pydantic import BaseModel, Field


class Video(BaseModel):
    """Database representation of a flagged video."""

    video_id: str
    channel_id: str | None = None
    title: str | None = None
    score: float = 0.0
    total_votes: int = 0
    locked: bool = False
    hidden: bool = False
    shadow_hidden: bool = False
    video_duration: float | None = None
    is_short: bool = False
    first_reported: datetime
    last_updated: datetime
    service: str = "youtube"


class VideoCategory(BaseModel):
    """Per-category vote aggregates for a video."""

    video_id: str
    category: str
    vote_count: int = 0
    weighted_score: float = 0.0


class CategoryDetail(BaseModel):
    """Vote count and weighted score for a single category."""

    votes: int
    weighted_score: float = Field(serialization_alias="weightedScore")


class VideoResponse(BaseModel):
    """API response for video lookups."""

    video_id: str = Field(serialization_alias="videoId")
    score: float
    categories: dict[str, CategoryDetail]
    total_votes: int = Field(serialization_alias="totalVotes")
    locked: bool
    channel_id: str | None = Field(default=None, serialization_alias="channelId")
    channel_score: float | None = Field(default=None, serialization_alias="channelScore")
    last_updated: datetime = Field(serialization_alias="lastUpdated")

    model_config = {"populate_by_name": True}

File: realtube-go/internal/model/video.go
package model

import "time"

// Video represents a flagged video in the database.
type Video struct {
	VideoID       string    `json:"videoId"`
	ChannelID     *string   `json:"channelId,omitempty"`
	Title         *string   `json:"title,omitempty"`
	Score         float64   `json:"score"`
	TotalVotes    int       `json:"totalVotes"`
	Locked        bool      `json:"locked"`
	Hidden        bool      `json:"-"`
	ShadowHidden  bool      `json:"-"`
	VideoDuration *float64  `json:"videoDuration,omitempty"`
	IsShort       bool      `json:"isShort,omitempty"`
	FirstReported time.Time `json:"firstReported"`
	LastUpdated   time.Time `json:"lastUpdated"`
	Service       string    `json:"service,omitempty"`
}

// VideoCategory represents per-category vote aggregates for a video.
type VideoCategory struct {
	VideoID       string  `json:"videoId"`
	Category      string  `json:"category"`
	VoteCount     int     `json:"votes"`
	WeightedScore float64 `json:"weightedScore"`
}

// VideoResponse is the API response for video lookups.
type VideoResponse struct {
	VideoID      string                       `json:"videoId"`
	Score        float64                      `json:"score"`
	Categories   map[string]*CategoryDetail   `json:"categories"`
	TotalVotes   int                          `json:"totalVotes"`
	Locked       bool                         `json:"locked"`
	ChannelID    *string                      `json:"channelId,omitempty"`
	ChannelScore float64                      `json:"channelScore,omitempty"`
	LastUpdated  time.Time                    `json:"lastUpdated"`
}

// CategoryDetail holds the vote count and weighted score for a single category.
type CategoryDetail struct {
	Votes         int     `json:"votes"`
	WeightedScore float64 `json:"weightedScore"`
}
